lib/stripe_mock/request_handlers/helpers/subscription_helpers.rb

```diff
@@ -56,6 +56,24 @@ def custom_subscription_params(plans, cus, options = {})
           params.merge!({status: 'trialing', current_period_end: end_time, trial_start: start_time, trial_end: end_time, billing_cycle_anchor: options[:billing_cycle_anchor] || created_time})
         end
 
+        # Support updating subscription with multiple plan items
```

Comment:
I cannot tell if this is something specific to our app, or if it's making up for a version skew in the API? Like we're using an older version of the API, which supported `{ plan: 'gold', quantity: 2 }`, but StripeMock didn't support that?

Reply:
Hey, sorry for any confusion here, @alexmamonchik. We have [a branch on a fork](https://github.com/docsend/stripe-ruby-mock/tree/stripe-ruby-mock-3.1.0.rc3) of stripe-ruby-mock which includes a number of "customizations" that were made over the past… many years. We need to upgrade our Stripe SDK, and that means we also need to update our StripeMock.

So, I rebased our custom branch atop the latest `init-v4` which had both `3.1.0` in it, and some other changes that we need (like newer Stripe SDK versions). My hope was to tease out all of the "customizations" which really should be things in our code, from any actual changes we needed/wanted to upstream into the mainline StripeMock.

This draft PR is really more for me to tell exactly where those changes are.

So for this specific change ("Support updating subscription with multiple plan items"), are you saying that current `3.1.0` already supports this?

Reply:
unfortunately  it's not about multiple items https://github.com/stripe-ruby-mock/stripe-ruby-mock/blob/master/spec/shared_stripe_examples/subscription_items_examples.rb#L35

Reply:
Ah, I've confirmed that this was actually upstreamed (my the original author of our custom patch) back in #528.

However, that upstreamed change is missing one thing our custom patch had: set the `subscription.quantity` to whatever the `subscription.items.data[0].quantity` was, so long as there was only a single item. This is technically in line with how the Stripe API appears to work, but it's also fragile, and presumably deprecated.

In any case, this custom patch can be removed too! Meaning our code is very close to in-line with what's currently on the upstream `init-v4` branch! And I have PRs open to close those remaining deltas too!